src/nvim/os/fs.c

```diff
@@ -503,3 +503,42 @@ bool os_fileid_equal_fileinfo(const FileID *file_id,
          && file_id->device_id == file_info->stat.st_dev;
 }
 
+
+/// Open directory for listing contents
+///
+/// @param scandir_req scandir request
+/// @param directory_path Directory path to be listed
+/// @return The number of directory entries
+int os_scandir_open(uv_fs_t *scandir_req, const char *directory_path)
```

Comment:
I think we should not expose LibUV types, what about doing something like `FileInfo` in `fs_defs.h`?